### DISCUSSION TITLE
Webp green pixels at the bottom of picture

### DISCUSSION THREAD

**Discussion by leventefabry-mof** [saucecontrol/PhotoSauce#112] [Q&A] [answered]
Hi!

I really appreciate your tool it's working fantastic. I have an issue with the .webp format and I hope you can help me.

I'm using the 0.13.2 version of MagicScaler and the 1.2.4-preview1 version of "PhotoSauce.NativeCodecs.Libwebp".

I registered the codecs at the start-up:
`CodecManager.Configure(codecs => { codecs.UseLibwebp(); });`

In the image process, the input is a byte array and the output is a stream which I convert back to a byte array.
When I try to set the encoder format in the settings at some image size I get a thin green pixel line at the bottom of the pic, as you can see. (I'm unable to attach .webp format so it's only a screenshot.)

![Screenshot 2023-04-27 093756](https://user-images.githubusercontent.com/104758603/234793048-01c38876-8e4b-4ff1-b313-b360d3cbe4ab.jpg)

I set the encoder format like this:
`settings.TrySetEncoderFormat(ImageMimeTypes.Webp);`
If I remove this code the picture will be rendered well, without the green line, although the image size will be bigger, which can be problematic for me. I want to keep the image size as low as I can.

These are my settings:

![Screenshot 2023-04-27 094528](https://user-images.githubusercontent.com/104758603/234794917-634e1ff3-7a86-455d-a640-2a17b6c846b4.jpg)
![Screenshot 2023-04-27 094544](https://user-images.githubusercontent.com/104758603/234794914-aec367a4-f89f-4ef1-9afd-6932bde2b305.jpg)

I would really appreciate it if you could give me any further ideas on what can I do to prevent this issue.

**Comment by saucecontrol** [comment]
Howdy!  Glad you're finding the library useful.

That output looks like a bug, possibly to do with the odd crop height.  I'd need a full repro to diagnose it, though.  Can you share your full code, environment info, and source image?

**Comment by leventefabry-mof** [comment]
If I send you the repo and the source image via email, is that work for you? Unfortunately, the real repo is inside an organization.
I use Windows 11 but the app is a function app, running on a Windows App service on Azure. I experience the same issue in both environments.

**Comment by leventefabry-mof** [comment]
I was able to reproduce the issue with this code snippet as well. It's a net7.0 console app. Windows 11, dotnet version: 7.0.203
```cs
using System.Drawing;
using PhotoSauce.MagicScaler;
using PhotoSauce.NativeCodecs.Libwebp;

CodecManager.Configure(codecs => { codecs.UseLibwebp(); });

var image = Image.FromFile("D:\\ImageProcessor\\cheetah.jpg");
using var ms = new MemoryStream();
image.Save(ms,image.RawFormat);
var bytes = ms.ToArray();

var settings = new ProcessImageSettings { Height = 465 };

settings.TrySetEncoderFormat(ImageMimeTypes.Webp);

using var outputStream = new MemoryStream();
MagicImageProcessor.ProcessImage(bytes, "D:\\ImageProcessor\\cheetah_output.webp", settings);
```

I attached here the original image:
![cheetah](https://user-images.githubusercontent.com/104758603/234838167-7b426436-dbdd-4b9d-9536-ceca92b7c9ab.jpg)

**Reply by saucecontrol** [reply, marked as answer]
Thanks!  I was able to reproduce the issue using your sample.

There is an off-by-1 bug in the encoder that's triggered by an image height that happens to be exactly one greater than a multiple of 8 (e.g. 465 = 58 * 8 + 1) under certain circumstances, leaving the last row of the image undefined.  The fix is simple, but I'll need to adjust my release strategy to get an updated package published.  Might take me a few days to figure out the best way to handle it.

**Reply by leventefabry-mof** [reply]
Thank you very much for your help, I really appreciate it. I'll look forward to the new version of the package.

**Reply by leventefabry-mof** [reply]
@saucecontrol Hi Clinton, any update on this?

**Reply by saucecontrol** [reply]
Yeah, I had to pause on this project for a bit, but I'll be pushing a new release hopefully by the end of July.
